trlx/orchestrator/ppo_orchestrator.py

```diff
@@ -206,21 +207,23 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
                     for ix in range(n)
                 ]
             else:
+                logprobs = logprobs_from_logits(logits[:, :-1, :], all_tokens[:, 1:])
+                ref_logprobs = logprobs_from_logits(
+                    ref_logits[:, :-1, :], all_tokens[:, 1:]
+                )
+
                 n = samples.shape[0]
-                values = values.cpu()
+                values = values.cpu()[:, :-1]
                 logprobs = logprobs.cpu()
                 ref_logprobs = ref_logprobs.cpu()
                 query_tensors = query_tensors.cpu()
                 response_tensors = response_tensors.cpu()
-                start = (
-                    query_tensors.shape[1] - 1
-                )  # left shift by 1 ref: https://github.com/lvwerra/trl/blob/main/trl/trainer/ppo_trainer.py#L425
-                ends = start + attention_mask[:, start:].sum(1) - 1
-                for ix in range(n):
-                    if ends[ix] == all_tokens.shape[1]:
-                        ends[ix] = ends[ix] - 1
-                all_values = [values[ix, start - 1 : ends[ix] - 1] for ix in range(n)]
+
+                start = query_tensors.shape[1] - 1
+                ends = start + attention_mask[:, start:].sum(1)
+                all_values = [values[ix, start : ends[ix]] for ix in range(n)]
```

Comment:
We probably should leave this out until we're 100% sure this is the right fix - we need unit tests for these.

Reply:
I do agree but we should have work version at least for simple tasks.

Reply:
Yeah, that's true. If this is just a revert of the stuff introduced in the T5 then we should keep it other I'll defer @LouisCastricato for second opinions.

Reply:
Leave for now, fix later